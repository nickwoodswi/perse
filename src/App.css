html{
    background: url(./images/gym.jpg) no-repeat center center fixed; 
    -webkit-background-size: cover;
    -moz-background-size: cover;
    -o-background-size: cover;
    background-size: cover;
    scroll-behavior: smooth;
}

header{
    display: flex;
    align-items: center;
    width: 100%;
    padding-left: 5px;
    background-color: white;
}
    header h1{
        font-size: 72px;
        font-style: italic;
        margin: 0;
        padding-top: -10;
    }
    #app-logo{
        height: 96px;
    }
@media all and (min-width: 660px) {
    nav{
        justify-content: flex-start !important
    }
}
nav{
    width: 100%;
    display: flex;
    justify-content: space-between;
    margin-top: 5px;
}
    .nav-button{
        display: flex;
        justify-content: center;
        align-items: center;
        text-align: center;
        width: 76px;
        height: 50px;
        background-color: white;
        box-shadow: 2px 0px 10px darkgrey;
        margin: 5px;
        font-size: 16px;
        padding: 10px;
        border-radius: 5px;
        cursor: pointer
    }
        a {
            text-decoration: none;
            color: black;
            font-weight: bold;
            font-style: italic;
        }
        .nav-button:active{
            background-color: red;
        }
.about-perse-container{
    width: 100%;
    display: flex;
    justify-content: center;
    margin-bottom: 25px;
}
    .about-perse{
        width: 100%;
        max-width: 600px;
        background-color: white;
        padding: 25px;
        margin: 5px;
    }
    .perse-summary{
        font-size: 32px;
    }
button{
    height: 36px;
    margin-left: 15px;
}
h4{
    margin-bottom: 0;
}
.athlete-selection{
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-top: 25px;
}
.select-dates{
    margin-top: 25px;
}
    .select-dates h3{
        color: rgba(255,255,255,0.9);
        font-size: 14px;
        font-style: italic;
    }
.add-exercise-types{
    display: flex;
    margin-bottom: 5px;
}
    .add-exercise-type{
        display: flex;
        align-items: center;
        justify-content: center;
        width: 142px;
        height: 75px;
        background-color: white;
        margin-right: 5px;
        border-radius: 5px;
        box-shadow: 2px 2px 2px darkgrey;
    }
        .add-exercise-type:active{
            background-color: red;
        }

#ex-name-input{
    width: 285px;
}
.rep-type{
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    width: 100%;
}
    .set-reps{
        width: 295px;
        margin-bottom: 15px;
    }
        #rep-type{
            width: 170px;
            height: 36px;
            border-radius: 5px;
            border: none;
            font-size: 16px;
            box-shadow: 2px 2px 2px darkgrey;
        }
            #rep-number{
                width: 115px;
                height: 34px;
                border-radius: 5px;
                border: none;
                margin-left: 2px;
                font-size: 16px;
                padding-left: 5px;
                box-shadow: 2px 2px 2px darkgrey;
            }

.rep-spec{
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    width: 100%;
}
    .add-weight{
        width: 145px;
        margin-right: 3px;
        color: rgba(255,255,255,.8);
        font-style: italic;
    }
        .add-weight input{
            border-radius: 5px;
            border: none;
            padding-left: 5px;
            font-size: 16px;
            box-shadow: 2px 2px 2px darkgrey
        }
    .rep-distance{
        width: 143px;
        color: rgba(255,255,255,0.9);
        font-style: italic;
    }
        .rep-distance input{
            border-radius: 5px;
            border: none;
            font-size: 16px;
            padding-left: 5px;
            box-shadow: 2px 2px 2px darkgrey;
            width: 135px !important;
        }
#weight-number{
    width: 140px;
    height: 30px;
    margin-bottom: 10px;
}
#rep-distance{
    width: 142px;
    height: 30px;
    margin-bottom: 10px;
}
.intensity-spec{
    display: flex;
    justify-content: center;
    width: 100%;
    margin-top: 10px;
}
    .tempo-spec{
        width: 150px;
        padding-left: 2px;
        font-style: italic;
        color: rgba(255,255,255,0.9)
    }
        .tempo-spec h4{
            margin: 0;
        }
    .subrest-spec{
        width: 145px;
        font-style: italic;
        color: rgba(255,255,255,0.9)
    }
        .subrest-spec h4{
            margin: 0;
        }
        .intensity-time-selector{
            width: 70px;
            height: 36px;
            border-radius: 5px;
            border: none;
            box-shadow: 2px 2px 2px darkgrey;
            font-size: 16px;
        }
        .intensity-time-input{
            width: 65px;
            height: 34px;
            border-radius: 5px;
            border: none;
            margin-left: 3px;
            box-shadow: 2px 2px 2px darkgrey;
        }
.rest-spec{
    display: flex;
    flex-wrap: wrap;
    width: 100%;
    max-width: 300px;
    align-items: flex-end;
}
    .rest-spec-header-container{
        width: 100%;
    }
    .rest-editor{
        width: 50%;
        display: flex;
        flex-wrap: wrap;
        padding-left: 5px;
        margin-top: 10px;
        color: rgba(255,255,255,0.9);
        font-style: italic
    }
        .rest-editor h3{
            margin: 0;
        }
        .rest-selector{
            width: 65px;
            height: 36px;
            margin: 0px;
            border-radius: 5px;
            border: none;
            box-shadow: 2px 2px 2px darkgrey;
        }
        .rest-input{
            width: 70px;
            height: 34px;
            margin: 0px;
            border-radius: 5px;
            border: none;
            margin-left: 5px;
            box-shadow: 2px 2px 2px darkgrey;
        }
    #add-exercise-button{
        background-color: white;
        margin-right: 13px;
        height: 60px;
        border-radius: 5px;
        border: none;
        box-shadow: 2px 2px 2px darkgrey;
    }
        #add-exercise-button:active{
            background-color: greenyellow
        }
.date-assignment-container{
    margin-bottom: 25px;
}
.date-type-selectors{
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
}
    .date-type-selector{
        display: flex;
        justify-content: center;
        align-items: center;
        width: 90px;
        height: 50px;
        background-color: white;
        font-size: 16px;
        margin-right: 5px;
        margin-top: 5px;
        border-radius: 5px;
        box-shadow: 2px 2px 2px darkgrey;
        
    }
        .date-type-selector:active{
            background-color: fuchsia;
        }

.App{
    display: flex;
    flex-wrap: wrap;
}
    .assignment-spec{
        display: flex;
        flex-wrap: wrap;
        flex-direction: column;
        width: 100%;
    }
        .build-exercise{
            width: 100%;
            display: flex;
            flex-wrap: wrap;
            flex-direction: column;
            align-items: center;
            max-width: 500px;
            margin-left: auto;
            margin-right: auto;
        }
            .build-exercise h3{
                margin-bottom: 5px;
            }
        .build-workout{
            display: flex;
            flex-wrap: wrap;
            justify-content: center;
            width: 100%;
            max-width: 525px;
            margin-left: auto;
            margin-right: auto;
            padding-right: 5px;
            background-color: rgba(0,0,0,.8);
            border-bottom-left-radius: 5px;
            border-bottom-right-radius: 5px;
        }
            .sets{
                display: flex;
                flex-direction: column;
                justify-content: center;
                flex-wrap: wrap;
                align-items: center;
                /* background-color: pink; */
                margin-left: auto;
                margin-right: auto;
                width: 100%;
                padding-bottom: 25px;
            }
            .assign-workout-button{
                display: flex;
                justify-content: center;
                align-items: center;
                width: 200px;
                height: 75px;
                margin-top: 5px;
                margin-bottom: 5px;
                background-color: yellow;
            }
            .workout{
                width: 100%;
                max-width: 530px;
                display: flex;
                flex-wrap: wrap;
                flex-direction: column;
                align-items: center;
                justify-content: flex-start;
                background-color: rgba(0,0,0,.8);
                border-top-left-radius: 5px;
                border-top-right-radius: 5px;
                margin-left: auto;
                margin-right: auto;
                margin-top: 25px;
            }
                .assign-workout-headline{
                    width: 100%;
                    display: flex;
                    justify-content: center;
                    background-color: white;
                    border-top-left-radius: 5px;
                    border-top-right-radius: 5px;
                }
                .workout-selector-container{
                    max-width: 380px;
                    padding-top: 15px;
                    padding-right: 15px;
                    padding-left: 15px;
                }
                    .add-workout{
                        display: flex;
                        justify-content: space-between;
                        align-items: center;
                        background-color: rgba(211, 211, 211, 0.95);
                        padding-bottom: 5px;
                        border-radius: 5px;
                    }
                    .athlete-selection{
                        display: flex;
                        flex-wrap: wrap;
                        justify-content: space-between;
                        align-items: center;
                        padding-bottom: 5px;
                    }
                        .athlete-selection-headline{
                            width: 100%;
                            max-width: 301px;
                            color: rgba(255, 255, 255, 0.9);
                            font-style: italic;
                            font-size: 12px;
                        }
                        .select-athlete{
                            width: 100%;
                            background-color: rgba(211, 211, 211, 0.9);
                            padding-bottom: 5px;
                            border-radius: 5px;
                        }
                    .workout-selector{
                        width: 120px;
                        height: 70px;
                    }
                    .build-workout-button{
                        display: flex;
                        width: 120px;
                        height: 70px;
                        justify-content: center;
                        align-items: center;
                        background-color: yellow;
                    }
                        .build-workout-button-text{
                            text-align: center;
                        }
                .workout h3{
                    margin-bottom: 5px;
                }
                .build-exercise{
                    /* background-color: green; */
                    display: flex;
                }
                    .display-exercise-builder{
                        height: 75px;
                        background-color: rgba(0,0,0,.5);
                        margin-bottom: 25px;
                        margin-top: 15px;
                        padding: 20px;
                        margin-left: auto;
                        margin-right: auto;
                        border-radius: 5px;
                        border: none;
                        width: 200px;
                        font-size: 16px;
                        color: fuchsia;
                        cursor: pointer;
                        font-weight: bold;
                    }
                    .exercise-builder-hidden{
                        display: none;
                    }
                    .exercise-builder{
                        width: 100%;
                        background-color: rgba(0,0,0,.8);
                        display: flex;
                        flex-wrap: wrap;
                        justify-content: center;
                        margin-left: auto;
                        margin-right: auto;
                        margin-bottom: 25px;
                        padding-top: 25px;
                        padding-bottom: 25px;
                        border-radius: 5px;
                    }
                .set-reps h4{
                    color: rgba(255,255,255,.9);
                    font-style: italic;
                }