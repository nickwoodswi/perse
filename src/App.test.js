import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders learn react link', () => {
  render(<MemoryRouter><App /></MemoryRouter>);
});

it('renders the UI as expected', () => {
  const tree = renderer
      .create(<MemoryRouter><App /></MemoryRouter>)
      .toJSON();
  expect(tree).toMatchSnapshot();
});
