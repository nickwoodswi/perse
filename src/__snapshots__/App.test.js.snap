// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <header>
    <div
      className="header"
    >
      <h1>
        PERSE
      </h1>
    </div>
    <img
      id="app-logo"
      src="hanging-knee-raises-crossfit-silhouette.png"
    />
  </header>,
  <nav>
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <div
        className="nav-button"
      >
        ABOUT
      </div>
    </a>
    <a
      aria-current={null}
      href="/assign-workout"
      onClick={[Function]}
    >
      <div
        className="nav-button"
      >
        ASSIGN
      </div>
    </a>
    <a
      aria-current={null}
      href="/view-workouts"
      onClick={[Function]}
    >
      <div
        className="nav-button"
      >
        VIEW
      </div>
    </a>
  </nav>,
  <main
    className="App"
  >
    <div
      className="about-perse-container"
    >
      <div
        className="about-perse"
      >
        <div
          className="perse-summary"
        >
          <p>
            <b>
              Perse
            </b>
             is an easy-to-use workout record and builder for coaches, athletes, and personal trainers.
          </p>
        </div>
        <p>
          Previously built workouts are stored in the Perse database for quick assignment. Select one, and assign it to an athlete on a single date, a range of dates, or on recurring dates.
        </p>
        <p>
          Trying something new or leveling up? Build a new workout by selecting and adding exercises from those already added to the database. Or, name and add your own exercise before setting intensity by weight, distance, speed, tempo, and/or rest.
        </p>
        <p>
          Review the programs you've set by viewing workouts assigned to any athlete in the system.
        </p>
        <p>
          Perse was originally built with ReactJS for Mike Woods, a former Olympic speedskater and current youth speedskating coach for U.S. Speedskating.
        </p>
        <p>
          <i>
            Â©2020 Nick Woods, 
            <a
              href="mailto:nick.woods.wi@gmail.com"
            >
              nick.woods.wi@gmail.com
            </a>
          </i>
        </p>
      </div>
    </div>
  </main>,
]
`;
